samples/DirectX/D3D12/HelloConstBuffer12.cs

```diff
@@ -0,0 +1,807 @@
+                        Flags = D3D12_DESCRIPTOR_HEAP_FLAGS.D3D12_DESCRIPTOR_HEAP_FLAG_SHADER_VISIBLE,
+                        Type = D3D12_DESCRIPTOR_HEAP_TYPE_CBV_SRV_UAV,
+                    };
+                    fixed (ID3D12DescriptorHeap** cbvHeap = &_cbvHeap)
+                    {
+                        iid = IID_ID3D12DescriptorHeap;
+                        ThrowIfFailed(nameof(ID3D12Device.CreateDescriptorHeap), _device->CreateDescriptorHeap(&cbvHeapDesc, &iid, (void**)cbvHeap));
+                    }
+                    _cbvDescriptorSize = _device->GetDescriptorHandleIncrementSize(D3D12_DESCRIPTOR_HEAP_TYPE_CBV_SRV_UAV);
+                }
+
+                // Create frame resources.
+                {
+                    var rtvHandle = _rtvHeap->GetCPUDescriptorHandleForHeapStart();
+
+                    // Create a RTV for each frame.
+                    iid = IID_ID3D12Resource;
+
+                    for (var n = 0u; n < FrameCount; n++)
+                    {
+                        ID3D12Resource* renderTarget;
+                        {
+                            ThrowIfFailed(nameof(IDXGISwapChain3.GetBuffer), _swapChain->GetBuffer(n, &iid, (void**)&renderTarget));
+                            _device->CreateRenderTargetView(renderTarget, pDesc: null, rtvHandle);
+                            rtvHandle.Offset(1, _rtvDescriptorSize);
+                        }
+                        _renderTargets[unchecked((int)n)] = renderTarget;
+                    }
+                }
+
+                fixed (ID3D12CommandAllocator** commandAllocator = &_commandAllocator)
+                {
+                    iid = IID_ID3D12CommandAllocator;
+                    ThrowIfFailed(nameof(ID3D12Device.CreateRenderTargetView), _device->CreateCommandAllocator(D3D12_COMMAND_LIST_TYPE_DIRECT, &iid, (void**)commandAllocator));
+                }
+            }
+            finally
+            {
+                if (debugController != null)
+                {
+                    debugController->Release();
+                }
+
+                if (factory != null)
+                {
+                    factory->Release();
+                }
+
+                if (adapter != null)
+                {
+                    adapter->Release();
+                }
+
+                if (swapChain != null)
+                {
+                    swapChain->Release();
+                }
+            }
+        }
+
+        // Load the sample assets.
+        private void LoadAssets()
+        {
+            Guid iid;
+            ID3DBlob* signature = null;
+            ID3DBlob* error = null;
+            ID3DBlob* vertexShader = null;
+            ID3DBlob* pixelShader = null;
+
+            try
+            {
+                // Create a root signature consisting of a descriptor table with a single CBV.
+                {
+                    D3D12_FEATURE_DATA_ROOT_SIGNATURE featureData = new D3D12_FEATURE_DATA_ROOT_SIGNATURE {
```

Comment:
The type is apparent based on `new D3D12_FEATURE_DATA_ROOT_SIGNATURE`:
```suggestion
                    var featureData = new D3D12_FEATURE_DATA_ROOT_SIGNATURE {
```